main.tf

```diff
@@ -0,0 +1,510 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~>5.0"
+    }
+  }
+}
+
+data "template_file" "main_tf" {
+  template = file("${path.module}/templates/main.tf.tpl")
+
+  vars = {
+    bucket_name             = var.s3_tf_bucket_name
+    region                  = var.aws_region
+    dynamodb_table          = var.dynamodb_tf_locks_name
+    security_group_id_input = var.security_group_id != null ? var.security_group_id : ""
+    public_subnet_id_input  = length(var.public_subnet_ids) > 0 ? element(var.public_subnet_ids, 0) : ""
+
+  }
+}
```

Comment:
See the deprecation notice [here](https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file.html). Use the [templatefile](https://developer.hashicorp.com/terraform/language/functions/templatefile) function instead. I can't use the module as it is now.

![image](https://github.com/frgrisk/terraform-aws-turbo-deploy/assets/16367391/2fd35dc4-21ba-489b-b094-9b0ad64338e8)